postproengine/wake_meandering.py

```diff
@@ -0,0 +1,391 @@
+import numpy as np
+import pandas as pd
+import os
+import pickle
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import importlib.util
+import sys
+
+# Add check for samwich package
+name = 'samwich'
+if (spec := importlib.util.find_spec(name)) is not None:
+    module = importlib.util.module_from_spec(spec)
+    sys.modules[name] = module
+    spec.loader.exec_module(module)
+    from samwich.dataloaders import PlanarData
+    from samwich.waketrackers import track
+    usesamwhich=True
+else:
+    print(f"can't find the {name!r} module")
+    usesamwhich=False
+
+"""
+Plugin for computing wake meandering statistics
+
+See README.md for details on the structure of classes here
+"""
+
+def get_wake_centers(u,YY,ZZ,method='ConstantArea',weighting=lambda u: np.ones_like(u),args=None):
+    datadict = {}
+    datadict['y'] = np.copy(YY[:,:].T)
+    y_grid_center = (datadict['y'][-1,0] + datadict['y'][0,0])/2.0 
+    datadict['z'] = np.copy(ZZ[:,:].T)
+    datadict['u'] = u[:,:,:,0].transpose(0,2,1).copy()
+    
+    wakedata = PlanarData(datadict)
+    wake = track(wakedata.sliceI(),method=method,verbose=True)
+    wake.remove_shear(method='fringe',Navg=u.shape[0])
+    wake.wake_tracked = False
+    if method=='Gaussian':
+        yc,zc = wake.find_centers(umin=None,sigma=args)
+    if method=='ConstantArea':
+        yc,zc = wake.find_centers(args,weighted_center=weighting)
+    if method=='ConstantFlux':
+        flux = lambda u,u_w: -u * u_w  # function arguments correspond to field_names
+        yc,zc = wake.find_centers(args,flux_function=flux,field_names=('u','u_tot'))
+    return wake, yc + y_grid_center ,zc
+
+
+@registerplugin
+class postpro_wakemeander():
+    """
+    Wake meandering postprocess 
+    """
+    # Name of task (this is same as the name in the yaml)
+    name      = "wake_meander"
+    # Description of task
+    blurb     = "Compute wake meandering statistics"
+    inputdefs = [
+        # -- Execute parameters ----
+        {'key':'iplane',       'required':False,  'default':[0,],
+         'help':'i-index of planes to postprocess', },
+        {'key':'name',     'required':True,  'default':'',
+         'help':'An arbitrary name',},
+        {'key':'ncfile',   'required':True,  'default':'',
+        'help':'NetCDF sampling files of cross-flow planes', },
+        {'key':'group',   'required':False,  'default':None,
+         'help':'Which group to pull from netcdf file', },
+        {'key':'trange',    'required':False,  'default':[],
+            'help':'Which times to postprocess', }, 
+        {'key':'yhub',    'required':False,  'default':None,
+            'help':'Lateral hub-height center', }, 
+        {'key':'zhub',    'required':False,  'default':None,
+            'help':'Vertical hub-height', }, 
+        {'key':'method',    'required':True,  'default':'ConstantArea',
+            'help':'Method for computing wake center', }, 
```

Comment:
One minor comment -- in the help part, add the possible choices for `method`: ConstantArea, Gaussian, ConstantFlux.  Partly for users to see what are the available options, but also because `method` seems to case-sensitive and I always forget what to capitalize in the strings.

Reply:
Done! Also added and error check for the method type